litex/soc/software/libbase/uart.c

```diff
@@ -110,6 +110,8 @@ void uart_init(void)
 
 	uart_ev_pending_write(uart_ev_pending_read());
 	uart_ev_enable_write(UART_EV_TX | UART_EV_RX);
+	if (irq_attach)
+		irq_attach(UART_INTERRUPT, (isr_t)uart_isr);
```

Comment:
Possibly need wrapper see https://github.com/enjoy-digital/litex/pull/1815#issuecomment-1783806810

Reply:
Maybe, or just change the function signature. I just avoided this initially to minimise diffs.